src/StakeRegistry.sol

```diff
@@ -124,6 +126,72 @@ contract StakeRegistry is VoteWeigherBase, StakeRegistryStorage {
         // }
     }
 
+    /**
+     * @notice Similar to updateStakes() but this enforces that the caller is updating for all registered operators
+     *         across all quorums
+     * @param operators are the addresses of all operators whose stake information is getting updated
+     *                  must not contain duplicates and must be sorted by ascending operatorId
+     * @dev For each operator loop, we check that the operatorId is > than the previous loops iteration operatorId to ensure
+     *      we aren't updating duplicate operators. This is because we want to check at the end that we updated all registered
+     *      operators in the quorum and we verify this with indexRegistry.totalOperatorsForQuorum(quorumNumber)
+     */
+    function updateStakesAllOperators(address[] calldata operators) external {
+        // for each quorum, loop through operators and see if they are a part of the quorum
+        // if they are, get their new weight and update their individual stake history and the
+        // quorum's total stake history accordingly
+        for (uint8 quorumNumber = 0; quorumNumber < quorumCount; ) {
+            OperatorStakeUpdate memory totalStakeUpdate;
+            uint256 numOperatorsUpdatedInQuorum = 0;
+            bytes32 prevOperatorId = bytes32(0);
+            // for each operator
+            for (uint i = 0; i < operators.length; ) {
+                bytes32 operatorId = registryCoordinator.getOperatorId(operators[i]);
+                uint192 quorumBitmap = registryCoordinator.getCurrentQuorumBitmapByOperatorId(operatorId);
```

Comment:
this will get the bitmap and operator id for operators a bunch more times than it needs too... what if we just create a `uint32[] memory numOperatorsUpdatedInQuorum` and we loop thru each operator

Reply:
Agreed, maybe something like `bytes32[][] operatorIdsPerQuorum` as the input params? An array of operatorIds per quorum.
For each quorum, do a require check for `indexRegistry.totalOperatorsForQuorum(quorumNumber)` and the operatorIds array for that quorum. For each operatorId, check that the quorum number is in the corresponding bitmap.